File: asp-net-app/WebEMSOF/component/db/Class_db_payment_proof_methods.pas
unit Class_db_payment_proof_methods;

interface

uses
  Class_db;

type
  TClass_db_payment_proof_methods = class(TClass_db)
  private
    { Private Declarations }
  public
    constructor Create;
    procedure BindListControl(target: system.object);
  end;

implementation

uses
  mysql.data.mysqlclient,
  system.web.ui.webcontrols;

constructor TClass_db_payment_proof_methods.Create;
begin
  inherited Create;
  // TODO: Add any constructor code here
end;

procedure TClass_db_payment_proof_methods.BindListControl(target: system.object);
var
  dr: mysqldatareader;
begin
  self.Open;
  ListControl(target).Items.Clear;
  ListControl(target).Items.Add(listitem.Create('-- Select --',''));
  //
  dr := mysqlcommand.Create
    ('SELECT code,description FROM payment_proof_method_code_description_map ORDER BY description',connection).ExecuteReader;
  while dr.Read do begin
    ListControl(target).Items.Add(listitem.Create(dr['description'].tostring,dr['code'].tostring));
  end;
  dr.Close;
  self.Close;
end;

end.

File: asp-net-app/WebEMSOF/component/db/Class_db_payment_proof_methods.cs
using MySql.Data.MySqlClient;
using System;

using System.Web.UI.WebControls;
using Class_db;

namespace Class_db_payment_proof_methods
{
    public class TClass_db_payment_proof_methods: TClass_db
    {
        //Constructor  Create()
        public TClass_db_payment_proof_methods() : base()
        {
            // TODO: Add any constructor code here

        }
        public void BindListControl(object target)
        {
            MySqlDataReader dr;
            Open();
            ((target) as ListControl).Items.Clear();
            ((target) as ListControl).Items.Add(new ListItem("-- Select --", ""));
            using var my_sql_command = new MySqlCommand("SELECT code,description FROM payment_proof_method_code_description_map ORDER BY description", connection);
            dr = my_sql_command.ExecuteReader();
            while (dr.Read())
            {
                ((target) as ListControl).Items.Add(new ListItem(dr["description"].ToString(), dr["code"].ToString()));
            }
            dr.Close();
            Close();
        }

    } // end TClass_db_payment_proof_methods

}
